setup.py

```diff
@@ -0,0 +1,317 @@
+        if not os.path.exists(self.build_temp):
+            os.makedirs(self.build_temp)
+
+        try:
+            subprocess.Popen("echo $CXX", shell=True, stdout=subprocess.PIPE)
+            subprocess.check_call([cmake, ext.sourcedir] + cmake_args,
+                                  cwd=self.build_temp, env=env)
+            subprocess.check_call(
+                [cmake, '--build', '.', '--target', 'install'] + build_args,
+                cwd=self.build_temp
+            )
+        except subprocess.CalledProcessError as exc:
+            print("Status : FAIL", exc.returncode, exc.output)
+            raise
+
+    @staticmethod
+    def _find_cmake():
+        for candidate in ['cmake', 'cmake3']:
+            try:
+                out = subprocess.check_output([candidate, '--version'])
+                cmake_version = LooseVersion(
+                    re.search(r'version\s*([\d.]+)', out.decode()).group(1))
+                if cmake_version >= '3.5.0':
+                    return candidate
+            except OSError:
+                pass
+
+        raise RuntimeError("Project requires CMake >=3.5.0")
+
+
+class Docs(Command):
+    description = "Generate & optionally upload documentation to docs server"
+    user_options = [("upload", None, "Upload to docs server")]
+    finalize_options = lambda self: None
+
+    def initialize_options(self):
+        self.upload = False
+
+    def run(self):
+        # The extensions must be created inplace to inspect docs
+        self.reinitialize_command('build_ext', inplace=1)
+        self.run_command('build_ext')
+        self.run_command('build_sphinx')
+        if self.upload:
+            self._upload()
+
+    def _upload(self):
+        pass
+
+
+def setup_package():
+    NRN_PY_ROOT = 'share/lib/python'
+    NRN_PY_SCRIPTS = os.path.join(NRN_PY_ROOT, 'scripts')
+    NRN_COLLECT_DIRS = ['bin', 'lib', 'include', 'share/nrn/lib/hoc', 'share/nrn/lib']
+
+    docs_require = []  # sphinx, themes, etc
+    maybe_rxd_reqs = ['numpy', 'Cython'] if RX3D else []
+    maybe_docs = docs_require if "docs" in sys.argv else []
+    maybe_test_runner = ['pytest-runner'] if "test" in sys.argv else []
+
+    py_packages = [
+        'neuron',
+        'neuron.neuroml',
+        'neuron.tests',
+        'neuron.rxd',
+        'neuron.crxd',
+        'neuron.gui2'
+    ] + ["neuron.rxd.geometry3d"] if RX3D else []
+
+    REL_RPATH = "@loader_path" if sys.platform[:6] == "darwin" else "$ORIGIN"
+
+    extension_common_params = defaultdict(list,
+        library_dirs=["build/cmake_install/lib"],
+        libraries=["nrniv", "nrnpython{}".format(sys.version_info[0])],
+    )
+
+    extensions = [CMakeAugmentedExtension(
+        "neuron.hoc",
+        ["src/nrnpython/inithoc.cpp"],
+        cmake_collect_dirs=NRN_COLLECT_DIRS,
+        cmake_flags=[
+            '-DNRN_ENABLE_CORENEURON=OFF',
+            '-DNRN_ENABLE_INTERVIEWS=OFF',
```

Comment:
I'm not sure about all the implications with wheel, but it would be nice to have -DNRN_ENABLE_INTERVIEWS=ON
along with -DIV_ENABLE_X11_DYNAMIC=ON at least for mac builds so that things will work even if xquartz is not installed. -DNRN_ENABLE_RX3D=ON needs cython but it does not take much time to compile as the default is
-DNRN_RX3D_OPT_LEVEL=0 .  -DNRN_ENABLE_MPI=ON should be accompanied by
-DNRN_ENABLE_MPI_DYNAMIC=ON so that mpi will work if it exists and be ignored if mpi does not exist on the user machine (we generally build with openmpi). Hopefully soon it will also be possible to have
-DNRN_ENABLE_CORENEURON=ON even if limited to intel processors.